service.go

```diff
@@ -356,6 +359,22 @@ func (svc *Service) ProcessInvoice(ctx context.Context, invoice *lnrpc.Invoice)
 	return nil
 }
 
+// check if we need to publish an invoice
+func shouldPublishInvoice(invoice *lnrpc.Invoice) (ok bool) {
+
+	//don't publish unsettled invoice
+	if invoice.State != lnrpc.Invoice_SETTLED {
+		return false
+	}
+	//if the invoice is keysend, it needs record 696969
+	//(invoices always have always at least one htlc in them)
+	recs := invoice.Htlcs[0].CustomRecords
+	if invoice.IsKeysend {
+		return recs[TLV_WALLET_ID] != nil
```

Comment:
can we add some logs that the invoices are ignored then we can monitor it and also can potentially debug what is going on.